floodlight/models/kinematics.py

```diff
@@ -0,0 +1,102 @@
+import warnings
+from dataclasses import dataclass
+
+import numpy as np
+
+from floodlight.utils.types import Numeric
+from floodlight import Pitch, XY
+from floodlight.core.property import PlayerProperty
+
+
+@dataclass
+class ModelDistance:
+    """Class for calculating distances of players on the pitch.
+
+    Parameters
+    __________
+    pitch: Pitch
+        Pitch object of the data. Should at least include information about the data's
+        unit. If pitch.unit is 'percent', length and width of the pitch are also needed.
+
+    """
+
+    pitch: Pitch = None
+    length: Numeric = None
+    width: Numeric = None
+    unit: str = None
+
+    def __init__(self, pitch):
+        self.unit = pitch.unit
+        if pitch.length is not None:
+            self.length = pitch.length
+        if pitch.width is not None:
+            self.width = pitch.width
+        if pitch.unit is not None:
+            self.unit = pitch.unit
+        else:
+            warnings.warn(
+                "No information about pitch unit. Coordinates are assumed to be in m!"
+            )
```

Comment:
I think it's not very Pythonic (and also not necessary imo) to do such an assumption - if there's no unit (which shouldn't happen, as it is necessary to instantiate a pitch object), just throw an ValueError

Reply:
Removed it, it now throws an AttributeError.